### ISSUE TITLE
Rule proposal: Don't import `util` Node.js core module directly

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#211]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2460%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/211)

This module is the perfect case for using destructuring / named imports, as it's just a bunch of unrelated functions.

## Fail

```js
const util = require('util');
```

```js
import util from 'util';
```

```js
import * as util from 'util';
```

## Pass

```js
const {promisify} = require('util');
```

```js
import {promisify} from 'util';
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $60.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($60.00)

### Submitted pull Requests
- [#232 #211 - Add `no-direct-import` rule](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/232)
- [#789 Add `import-style` rule](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/789)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $60.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/211)

**Comment by futpib** [comment]
Instead of being `utils`-specific why not make it accept a set of modules as an option (with `utils` being among defaults)?

**Comment by sindresorhus** [comment]
Good idea! Let’s do that. We could add `lodash` and `underscore` to the default list too. Maybe we could also have a list for the inverse? For example, `chalk` should not be destructured, but people still do.

**Comment by arshadkazmi42** [comment]
@sindresorhus I would like to work on this.
Can you provide some more details what is the finalize approach for this?

**Comment by sindresorhus** [comment]
If anyone wants to work on this, see the previous attempt and feedback: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/232

**Comment by futpib** [comment]
Working on it
